rotkehlchen/db/dbhandler.py

```diff
@@ -630,36 +630,39 @@ def set_settings(self, write_cursor: 'DBCursor', settings: ModifiableDBSettings)
         )
         CachedSettings().update_entries(settings)
 
-    @overload
-    def get_cache(self, cursor: 'DBCursor', name: DBCache) -> Timestamp:
-        ...
-
-    @overload
-    def get_cache(self, cursor: 'DBCursor') -> dict[DBCache, Timestamp]:
-        ...
+    def get_cache_for_api(self, cursor: 'DBCursor') -> dict[str, int]:
+        """Returns a few key-value pairs that are rendered in the API
+        from the `key_value_cache` table of the DB. Defaults to `Timestamp(0)` if not found"""
+        cursor.execute(
+            'SELECT name, value FROM key_value_cache WHERE name IN (?,?);',
+            (DBCache.LAST_DATA_UPLOAD_TS.value, DBCache.LAST_BALANCE_SAVE.value),
+        )
+        db_cache = {name: int(value) for name, value in cursor}
+        return {  # Return with default value, if needed
+            DBCache.LAST_BALANCE_SAVE.value: db_cache.get(DBCache.LAST_BALANCE_SAVE.value, 0),
+            DBCache.LAST_DATA_UPLOAD_TS.value: db_cache.get(DBCache.LAST_DATA_UPLOAD_TS.value, 0),
+        }
 
     def get_cache(
             self,
             cursor: 'DBCursor',
-            name: DBCache | None = None,
-    ) -> dict[DBCache, Timestamp] | Timestamp:
-        """Returns the cache values from the `key_value_cache` table of the DB
-        - If `name` is not passed, it returns all the values in the table
-        - If `name` is passed, it returns the value of the given name, defaults to Timestamp(0)"""
-        if name is None:
-            cursor.execute('SELECT name, value FROM key_value_cache;')
-            return {
-                DBCache.deserialize(q[0]): Timestamp(int(q[1]))
-                for q in cursor
-            }
-        value = cursor.execute('SELECT value FROM key_value_cache WHERE name=?;', (name.value,)).fetchone()  # noqa: E501
-        return Timestamp(0) if value is None else Timestamp(int(value[0]))
-
-    def set_cache(self, write_cursor: 'DBCursor', cache: dict[DBCache, Timestamp]) -> None:
+            name: DBCache | str,
```

Comment:
Where do you need pure `str`? Isn't it more error prone? You have gone to the trouble of creating the `DBCache` enum, why not enforce it all around? This would:

1. Make it much less errro prone, as it would always get a valid value
2. Make it more performant as it would remove the if check you have in there